src/NzbDrone.Core/Datastore/ConnectionStringFactory.cs

```diff
@@ -74,5 +86,55 @@ private DatabaseConnectionInfo GetPostgresConnectionString(string dbName)
 
             return new DatabaseConnectionInfo(DatabaseType.PostgreSQL, connectionBuilder.ConnectionString);
         }
+
+        private DatabaseConnectionInfo GetPostgresConnectionInfoFromConnectionString(string connectionString)
+        {
+            var connectionBuilder = new NpgsqlConnectionStringBuilder(connectionString)
+            {
+                Enlist = false
+            };
+
+            return new DatabaseConnectionInfo(DatabaseType.PostgreSQL, connectionBuilder.ConnectionString);
+        }
+
+        private enum ConnectionStringType
+        {
+            SQLite,
+            PostgreSQLVars,
+            PostgreSQLConnectionString
+        }
```

Comment:
Move to the bottom of the class and update names to match of naming standard.

```suggestion
        private enum ConnectionStringType
        {
            Sqlite,
            PostgreSqlVars,
            PostgreSqlConnectionString
        }
```